src/test/java/uk/gov/hmcts/reform/hmc/service/HearingManagementServiceTest.java

```diff
@@ -1034,6 +1072,85 @@ void updateHearingRequestShouldThrowErrorWhenHearingIdNotPresentInDB() {
+        // assertEquals(versionNumber, hearingRequest.getVersionNumber());
+    }
+
+    void updateHearingShouldIncrementVersionNumber(Long hearingId, UpdateHearingRequest hearingRequest) {
+        final int versionNumber = hearingRequest.getRequestDetails().getVersionNumber();
+        when(hearingRepository.existsById(hearingId)).thenReturn(true);
+        when(caseHearingRequestRepository.getVersionNumber(hearingId)).thenReturn(versionNumber);
+        when(hearingRepository.getStatus(hearingId)).thenReturn(PutHearingStatus.HEARING_REQUESTED.name());
+
+        HearingEntity hearingEntity = generateHearingEntity(hearingId, PutHearingStatus.HEARING_REQUESTED.name());
+        when(hearingRepository.findById(hearingId)).thenReturn(Optional.of(hearingEntity));
+        when(hearingRepository.save(hearingEntity)).thenReturn(hearingEntity);
+
+        HearingResponse hearingResponse = hearingManagementService.updateHearingRequest(hearingId, hearingRequest);
+        // Check that version number has been incremented
+        assertEquals((versionNumber + 1), hearingResponse.getVersionNumber());
+    }
+
+    @Test
+    void updateHearingShouldIncrementVersionNumberBy1() {
+        final long hearingId = 2000000000L;
+        UpdateHearingRequest hearingRequest = TestingUtil.updateHearingRequest();
+        updateHearingShouldIncrementVersionNumber(hearingId, hearingRequest);
+    }
+
+    @Test
+    void updateHearingFailsByIncrementingVersionNumberByMoreThan1() {
+        final long hearingId = 2000000000L;
+        UpdateHearingRequest hearingRequest = TestingUtil.updateHearingRequest();
+        final int versionNumber = hearingRequest.getRequestDetails().getVersionNumber();
+
+        updateHearingShouldIncrementVersionNumber(hearingId, hearingRequest);
+        updateHearingShouldIncrementVersionNumber(hearingId, hearingRequest);
+
+        // Check that version number has NOT been incremented by 1
+        assertNotEquals((versionNumber + 1), hearingRequest.getRequestDetails().getVersionNumber());
+    }
+
+    @Test
+    void updateHearingValidationFailureShouldNotIncrementVersionNumber() {
+        UpdateHearingRequest hearingRequest = TestingUtil.updateHearingRequest();
+        final int versionNumber = hearingRequest.getRequestDetails().getVersionNumber();
+        // Expect validation error for invalid hearing window
+        hearingRequest.getHearingDetails().getHearingWindow().setFirstDateTimeMustBe(null);
+        hearingRequest.getHearingDetails().getHearingWindow().setHearingWindowStartDateRange(null);
+        hearingRequest.getHearingDetails().getHearingWindow().setHearingWindowEndDateRange(null);
+        assertThrows(BadRequestException.class, () -> hearingManagementService.updateHearingRequest(
+            2000000000L, hearingRequest));
+        // Check that version number remains the same
+        // assertEquals(versionNumber, hearingRequest.getRequestDetails().getVersionNumber());
```

Comment:
uncomment the assertEquals statement as the test has to confirm the version remains the same.

Reply:
Code fixed to allow assert to succeed